### DISCUSSION TITLE
Welcome to the Discussions board!

### DISCUSSION THREAD

**Discussion by jaywonchung** [ml-energy/leaderboard-v2#10] [Announcements]
## 👋 Welcome!

Here, you can:
- ask questions about the leaderboard,
- propose and discuss new ideas that improve the leaderboard,
- suggest new models, techniques, or software that are worth benchmarking,
- discuss what's on the leaderboard,
- and more!

Also, please be nice to everyone 🤗

**Comment by koenvanderleij** [comment]
Hi! I would like to cite your leaderboard in the introduction of a paper I am currently writing. So far I was not able to find the preferred citation on the website and the GitHub Repo. Could you let me know how you would like the leaderboard/project to be cited, and could you add this information to the about page, similar to how hugging face has included their preferred citation here https://huggingface.co/spaces/optimum/llm-perf-leaderboard? 
Thanks, and great work!

**Reply by jaywonchung** [reply]
Hi @koen430, thanks for the question. Do you need citation information ASAP, or can we take some time (a week or two) to discuss internally?

**Reply by koenvanderleij** [reply]
Hi @jaywonchung, there is no hurry! Take your time and I will update my citation when you have discussed it internally and decided on the citation preference.

**Reply by jaywonchung** [reply]
Hi @koen430, we just added our citation information at the bottom of our README and Leaderboard page. Thanks!

**Comment by vtlynch** [comment]
Hi,

I was looking for more information about the input tokens (prompt length) used for your LLM Chat testing?

In your About you cite using "500 human prompts from ShareGPT". As that dataset contains many thousands of conversations, Im assuming you picked some sort of representative sample.

Would you be able to share any information on the 500 prompts you chose? Such as the token average of the input prompts?

Thank you, 

Vincent

**Reply by jaywonchung** [reply]
Hey @vtlynch,
please see: https://github.com/ml-energy/leaderboard/tree/master/benchmark/llm_text_generation/chat/sharegpt

**Comment by ngohoanhkhoa** [comment]
Hello,
I’m currently working on reproducing the leaderboard for other LLMs using the script located at benchmark/llm_text_generation/code/scripts/aggregate_leaderboard_data.py. My setup is similar to yours (using a p4d.24xlarge instance).

I have a few questions:

1. After running the experiments, I noticed that the +stats.json files are missing — I only have the +results.json files. Could you please advise on how to generate or obtain the +stats.json files?
2. In the script aggregate_leaderboard_data.py, does the variable ss_total_energy correspond to the server_side_total_energy field in the results.json files?
3. Could you confirm whether the "Energy/req (J)" metric shown in the leaderboard corresponds to the server_side_energy_per_request field in the results.json?

Thank you very much for your support.
Best regards,

**Reply by jaywonchung** [reply]
Hi @ngohoanhkhoa.
1. The `+stats.json` file is a custom result file we generate from our fork of vLLM. I believe Docker images [here](https://hub.docker.com/r/mlenergy/vllm/tags) should work. But I do want to note that we're working on completely revamping and updating the benchmark/leaderboard, this time without forking vLLM. We hope to get it out in one to two months.
2. `ss_total_energy` stands for steady state total energy, which is described in [our paper](https://arxiv.org/abs/2505.06371).
3. At a high level, `ss_total_energy` together with the total number of tokens generated during the steady state is used to derive the average energy consumption of a single token during the server's steady state and then multiplied with each request's response length to get the response's energy consumption. That's `Energy/req (J)`.

**Reply by ngohoanhkhoa** [reply]
@jaywonchung Thank you for your response!
I use the mlenergy/vllm:v0.4.2-openai as in your tutorial. Would you mind sharing the code or approach you use to generate the +stats.json files from vLLM? I’d really appreciate any guidance you can provide.

**Reply by jaywonchung** [reply]
Can you try with `mlenergy/vllm:0.5.4-openai`? It's generated inside by the vLLM engine. But, just in case it doesn't work, we do not plan to fix it. We are working on a complete overhaul (v3.0) that will hopefully be much easier to run.

**Reply by ngohoanhkhoa** [reply]
@jaywonchung Is it this file ~/.config/vllm/usage_stats.json, as referenced in the documentation here: https://docs.vllm.ai/en/v0.5.2/serving/usage_stats.html ?
